feat: Реализовано ТЗ спринта 13

HotSauceVL opened the pull request:
feat: Добавлена структура проекта
feat: Реализованы контроллер, сервис, дао для оъектов Item и User

HotSauceVL committed:
feat: Реализовано ТЗ спринта 13
feat: Добавлена структура проекта
feat: Реализованы контроллер, сервис, дао для оъектов Item и User

avfyodorov commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,49 @@
 @RestController
 @RequestMapping("/items")
+@RequiredArgsConstructor
 public class ItemController {
```
Это не обязательно, на Ваше усмотрение, но я бы добавил и удаление предмета.

avfyodorov commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,49 @@
+import javax.validation.Valid;
+import java.util.List;
+
+@Validated
```
Совсем не нашёл нигде логирования, ни в контроллерах, ни в сервисе, ни в репозитарии. Лучше всё-таки добавить в одном из слоёв..

avfyodorov reviewed (commented):
Добрый вечер!
Работа очень хорошая, всего пара мелких замечаний. 
Да, и ещё, тесты надо пройти.

HotSauceVL committed:
feat: Исправлены замечания
feat: Реализовано удаление предмета через соответствующий эндпоинт
feat: Добавленно логгирование слоя Service
Style: Код исправлен по CodeStyle, тесты должны пройти=)

HotSauceVL committed:
Style: Код исправлен по CodeStyle, следующая попытка=)

HotSauceVL committed:
Style: Код исправлен по CodeStyle, может так=)

avfyodorov reviewed (commented):
Добрый день!

Замечаний нет.
Работа принята.
